File: FrontEnd/lib/domain/interfaces/usecases/weather/i_get_current_weather_use_case.dart
/// @file i_get_current_weather_use_case.dart
/// @description Interfaz del caso de uso para obtener el clima actual.
/// @module Weather
/// @layer Domain
library;

import '../../../../domain/entities/weather_data.dart';

abstract interface class IGetCurrentWeatherUseCase {
  /// Devuelve el clima actual para [location] (ciudad o coordenadas).
  ///
  /// [throws] AppError.network si no hay conexión y no hay caché válido.
  Future<WeatherData> execute(String location);
}

File: BackEnd/src/domain/interfaces/usecases/weather/IGetCurrentWeatherUseCase.ts
/**
 * @file IGetCurrentWeatherUseCase.ts
 * @description Interfaz del caso de uso para obtener el clima actual de una ubicación.
 * @module Weather
 * @layer Domain
 */

import type { WeatherQueryRequestDTO } from '../../../dtos/weather/weather-query-request.dto.js';
import type { WeatherResponseDTO } from '../../../dtos/weather/weather-response.dto.js';

export interface IGetCurrentWeatherUseCase {
  /**
   * @param dto — Coordenadas de la localización (lat, lon).
   * @returns WeatherResponseDTO con las condiciones actuales.
   * @throws {ValidationException} Si las coordenadas son inválidas.
   * @throws {ExternalServiceException} Si la API no está disponible.
   */
  execute(dto: WeatherQueryRequestDTO): Promise<WeatherResponseDTO>;
}
